src/services/ban_service.py

```diff
@@ -9,60 +9,62 @@
-                f"You are being banned from NA Ultimate Raiding - FF XIV for the following reason: \n> {reason} "
-                "\nYou may appeal this ban by contacting the moderators of the server in 30 days.",
-            )
-            logger.info(f"Successfully sent dm to {user.mention}")
-            dm_state = True
-        except Exception as e:
-            logger.error(f"DM to {user.mention} due failed due to error: {e}")
-            dm_state = False
+    if len(reason) >= 512:
+        return (
+            False,
+            False,
+            f"Unable to ban {user.mention}: reason is too long (above 512 characters). Please shorten the ban reason.",
+        )
 
-        # attempt Ban
-        try:
-            await user.ban(reason=reason)
-            logger.info(f"Successfully banned {user.mention}")
-            ban_result = True
-        except Exception as e:
-            logger.error(f"Ban of {user.mention} failed due to error: {e}")
-            ban_result = False
+    # Ensure invoking_member has a higher role position than the target user.
+    if user.top_role >= invoking_member.top_role:
+        return (
+            False,
+            False,
+            f"Unable to ban {user.mention}: You cannot ban a user with an equal or higher role than yourself.",
+        )
 
-        if not ban_result and dm_state:  # ban fail dm succeed.
-            result_description = f"Unable to ban {user.mention}, please ban via discord built-in ban feature. A DM has been sent to {user.mention} with ban reason."
-        elif ban_result and not dm_state:  # ban succeed dm fail.
-            result_description = (
-                f"Successfully banned {user.mention} but DM failed to send."
-            )
-        elif ban_result and dm_state:  # full success
-            result_description = f"Successfully banned {user.mention} and DM has been sent with ban reason."
-        else:  # full failure
-            result_description = f"Unable to ban {user.mention} and unable to send DM. Please ban via discord built-in ban feature or try again later."
+    dm_state = False
+    try:
+        await send_dm(
+            user,
+            f"You are being banned from NA Ultimate Raiding - FFXIV for the following reason:\n> {reason}\n"
+            "You may appeal this ban by contacting the moderators in 30 days.",
+        )
+        dm_state = True
+    except Exception as e:
+        logger.error(f"Failed to send DM to {user.mention}: {e}")
 
-    else:
-        # reason too large, ban action canceled.
-        result_description = f"Unable to ban: {user.mention}, reason given is too long (above 512 characters). Please shorten ban reason."
-        ban_result = False
-    # always return this.
-    return (ban_result, dm_state, result_description)
+    try:
+        # Delete messages only if delete_messages is True
+        # 604800 seconds is the maximum value for delete_message_seconds, and is equivalent to 7 days.
```

Comment:
can we describe a little more in detail why we're doing this in here (just for people who read this in the future)

Reply:
done